RadarSDK/RadarLocationManager.m

```diff
@@ -830,39 +830,71 @@ - (void)sendLocation:(CLLocation *)location stopped:(BOOL)stopped source:(RadarL
     self.sending = YES;
 
     RadarTrackingOptions *options = [Radar getTrackingOptions];
-    if (options.beacons) {
+    
+    void (^callTrackAPI)(NSArray<RadarBeacon *> *_Nullable) = ^(NSArray<RadarBeacon *> *_Nullable beacons) {
+        [[RadarAPIClient sharedInstance] trackWithLocation:location
+                                                   stopped:stopped
+                                                foreground:[RadarUtils foreground]
+                                                    source:source
+                                                  replayed:replayed
+                                                   beacons:beacons
+                                         completionHandler:^(RadarStatus status, NSDictionary *_Nullable res, NSArray<RadarEvent *> *_Nullable events, RadarUser *_Nullable user,
+                                                             NSArray<RadarGeofence *> *_Nullable nearbyGeofences, RadarConfig *_Nullable config, NSString *_Nullable token) {
+            self.sending = NO;
+
+            [self updateTrackingFromMeta:config.meta];
+            [RadarSettings setFeatureSettings:config.meta.featureSettings];
+            [self replaceSyncedGeofences:nearbyGeofences];
+        }];
+    };
+    
+    if (options.beacons &&
+        source != RadarLocationSourceBeaconEnter &&
+        source != RadarLocationSourceBeaconExit &&
+        source != RadarLocationSourceMockLocation &&
+        source != RadarLocationSourceManualLocation) {
+        
         [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:@"Searching for nearby beacons"];
-
-        if (source != RadarLocationSourceBeaconEnter && source != RadarLocationSourceBeaconExit && source != RadarLocationSourceMockLocation &&
-            source != RadarLocationSourceManualLocation) {
-            [[RadarAPIClient sharedInstance]
-                searchBeaconsNear:location
-                           radius:1000
-                            limit:10
-                completionHandler:^(RadarStatus status, NSDictionary *_Nullable res, NSArray<RadarBeacon *> *_Nullable beacons, NSArray<NSString *> *_Nullable beaconUUIDs) {
-                    if (beaconUUIDs && beaconUUIDs.count) {
-                        [self replaceSyncedBeaconUUIDs:beaconUUIDs];
-                    } else if (beacons && beacons.count) {
-                        [self replaceSyncedBeacons:beacons];
-                    }
+        
+        [[RadarAPIClient sharedInstance]
+         searchBeaconsNear:location
+         radius:1000
+         limit:10
```

Comment:
not yet necessarily, but I think at some point I think we'll want this limit set by a `featureSetting` so that we can choose how many beacons & how many geofences we sync down. (Might not matter if we're merely ranging beacons and not storing them as `CLRegion`s.